mne/channels/channels.py

```diff
@@ -336,6 +339,55 @@ def rename_channels(info, mapping):
     # it doesn't work)
     info['ch_names'] = [c['ch_name'] for c in chs]
 
+    # assert channel names are unique
+    assert len(info['ch_names']) == np.unique(info['ch_names']).shape[0]
+
+
+def define_sensor(info, mapping):
+    """Define the sensor type of channels.
+
+    Note: The following sensor types are accepted:
+        ecg, eeg, emg, eog, exci, ias, meg, mcg,
+        misc, ref_meg, resp, seeg, stim, syst
+
+    Parameters
+    ----------
+    info : dict
+        Measurement info.
+    mapping : dict
+        a dictionary mapping a channel to a sensor type (str)
+        {'EEG061': 'eog'}.
+    """
+    human2fiff = {'ecg': FIFF.FIFFV_ECG_CH,
+                  'eeg': FIFF.FIFFV_EEG_CH,
+                  'emg': FIFF.FIFFV_EMG_CH,
+                  'eog': FIFF.FIFFV_EOG_CH,
+                  'exci': FIFF.FIFFV_EXCI_CH,
+                  'ias': FIFF.FIFFV_IAS_CH,
+                  'meg': FIFF.FIFFV_MEG_CH,
+                  'mcg': FIFF.FIFFV_MCG_CH,
+                  'misc': FIFF.FIFFV_MISC_CH,
+                  'ref_meg': FIFF.FIFFV_REF_MEG_CH,
+                  'resp': FIFF.FIFFV_RESP_CH,
+                  'seeg': FIFF.FIFFV_SEEG_CH,
+                  'stim': FIFF.FIFFV_STIM_CH,
+                  'syst': FIFF.FIFFV_SYST_CH}
+
+    ch_names = info['ch_names']
+
+    # first check and assemble clean mappings of index and name
+    for ch_name, ch_type in mapping.items():
+        if ch_name not in ch_names:
+            raise ValueError("This channel name (%s) doesn't exist in info."
+                             % ch_name)
+
+        c_ind = ch_names.index(ch_name)
+        if ch_type not in human2fiff:
+            raise ValueError('This function cannot change to this '
+                             'channel type: %s.' % ch_type)
+    # Set sensor type
+        info['chs'][c_ind]['kind'] = human2fiff[ch_type]
```

Comment:
also, I went through and listed all possible ch types. I didn't see a
reason to restrict which one could be used.

-teon

On Tue, Apr 7, 2015 at 3:05 PM, Denis A. Engemann <notifications@github.com>
wrote:

> In mne/channels/channels.py
> <https://github.com/mne-tools/mne-python/pull/1932#discussion_r27875156>:
>
> > +                  'syst': FIFF.FIFFV_SYST_CH}
> > +
> > +    ch_names = info['ch_names']
> > +
> > +    # first check and assemble clean mappings of index and name
> > +    for ch_name, ch_type in mapping.items():
> > +        if ch_name not in ch_names:
> > +            raise ValueError("This channel name (%s) doesn't exist in info."
> > +                             % ch_name)
> > +
> > +        c_ind = ch_names.index(ch_name)
> > +        if ch_type not in human2fiff:
> > +            raise ValueError('This function cannot change to this '
> > +                             'channel type: %s.' % ch_type)
> > +    # Set sensor type
> > +        info['chs'][c_ind]['kind'] = human2fiff[ch_type]
>
> I think that's it. We have a different function for setting positions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/1932/files#r27875156>.
>